src/cswinrt/code_writers.h

```diff
@@ -457,8 +457,23 @@ namespace cswinrt
 
     void write_projection_arg(writer& w, method_signature::param_t const& param)
     {
-        w.write("%",
-            bind<write_parameter_name>(param));
+        switch (get_param_category(param))
+        {
+        case param_category::in:
+        case param_category::ref:
+            w.write("%", bind<write_parameter_name>(param));
+            break;
+        case param_category::out:
+            w.write("out %", bind<write_parameter_name>(param));
+            break;
+        case param_category::pass_array:
+        case param_category::fill_array:
```

Comment:
are these used when passing the parameter or declaring the parameter type?

Reply:
When passing the arguments.
in `event.Invoke(...)`

Reply:
so that would generate something like param[] which you might want to confirm whether is right.  When a type is being written for a method parameter, that would make sense like type[].  Look at write_parameter_name_with_modifier vs write_projection_parameter_type.

Reply:
That makes sense. The square brackets seem unnecessary in that case.